### DISCUSSION TITLE
MissionControl 0.6.1

### DISCUSSION THREAD

**Discussion by ndeadly** [ndeadly/MissionControl#384] [General]
<p align="left">
<img alt="GitHub Releases" src="https://img.shields.io/github/downloads/ndeadly/MissionControl/v0.6.1/total">
</p>

MissionControl 0.6.1

This is the tenth official release of Mission Control.

Just a small update to add support for the latest HOS (13.1.0) and Atmosphère (1.2.1) versions, and a couple of minor fixes. See below for the full changelog.

Valid analog stick deadzone parameters are now stored in virtual SPI flash file for each controller. This resolves an issue reported by several users where some controllers were experiencing analog stick drift even after calibration due to incorrect deadzone parameters being applied. Please delete the controller entries under `sdmc:/config/MissionControl/controllers/` so that these files can be recreated with the correct parameters. **_Note that you will need to redo any user calibrations after deleting these entries._**

Support for manually disabling a vendor command (SetTsi) for incompatible controllers has been added. Allowing the console to send the command to such controllers causes them to be disconnected shortly after connecting. In most cases, sending of this command can be disabled in code after the incompatibility has been reported. Some incompatible third-party controllers, however, are indistinguishable from the originals that _do_ support the command and must be flagged manually. To do so, a blank file `settsi_disable.flag` can be created under the directory for a specific controller under `sdmc:/config/MissionControl/controllers/<controller address>`. **_Note that this command is used for managing bandwidth between bluetooth devices and also the Wifi chip. It is therefore desirable to have the console send it where possible and should only be disabled where absolutely necessary._**

New users landing on this release page should first check out the readme on the [main project page](https://github.com/ndeadly/MissionControl) for the official project documentation. There you can find installation and usage instructions along with an FAQ section that will answer most of your questions.

***Atmosphère >= 1.2.0 is required to run this release.***

### Important notice for users of Xbox One controllers
It has been confirmed that the new Xbox Insiders controller firmware update will convert your controller to use Bluetooth LE, which is not currently supported. Please do not update if you wish to continue using your controller with Mission Control

### Changelog:
- Added bluetooth ips patches for 13.1.0.
- Updated module for libstratosphere refactor (thanks @SciresM)
- Added valid deadzone information to virtual SPI flash files (thanks @Banz99). 
- Added means of manually disabling sending of SetTsi vendor command to controllers that don't support it.

<hr /><em>This discussion was created from the release <a href='https://github.com/ndeadly/MissionControl/releases/tag/v0.6.1'>MissionControl 0.6.1</a>.</em>

**Comment by everyonesweird** [comment]
Thanks So much.

**Comment by everyonesweird** [comment]
seems like my ps4 controller while connected as 2nd player vibrates all the time like crazy. like even when I'm on home screen it just vibrates nonstop. But if connected as player 1, vibration stops and when I connect the joycon as 2nd, then ps4 controllers as 1st player again start to vibrate.
Also, my knock of ps4 controller does not connect. it gets detected but as soon as it connects, it switch offs.
It was fine on previous build where I could connect all 3 perfectly.

**Reply by ndeadly** [reply]
Weird. This has been an issue I observed with Wii/WiiU controllers for a while now, but I never saw it with any other controllers. I'm not really sure why it happens, it seems the console is spamming rumble packets in some menus/games under certain conditions. I haven't touched anything rumble related in the past few releases, so something else must have changed. The rumble code could definitely use some more attention though.

> Also, my knock of ps4 controller does not connect. it gets detected but as soon as it connects, it switch offs.
> It was fine on previous build where I could connect all 3 perfectly.

This is exactly what the `settsi_disable.flag` mentioned in the release notes above is for. Your knockoff controller will be using a different bluetooth chipset that doesn't like the SetTsi vendor command sent by the console. In older versions I was disabling this command for _all_ non-nintendo controllers, so your controller would have been happy. It turned out that faking the success of this command could create issues with input delays and degradation of WiFi speeds, so now I only disable it when a controller doesn't support it. In the case of DS4 controllers, the authentic ones _do_ support the command so we want to send it, but the third party controllers that don't support the command are indistinguishable from the real ones in software so we need to flag them manually.

**Reply by Zohan1994** [reply]
> Weird. This has been an issue I observed with Wii/WiiU controllers for a while now, but I never saw it with any other controllers. I'm not really sure why it happens, it seems the console is spamming rumble packets in some menus/games under certain conditions. I haven't touched anything rumble related in the past few releases, so something else must have changed. The rumble code could definitely use some more attention though.
> 
> > Also, my knock of ps4 controller does not connect. it gets detected but as soon as it connects, it switch offs.
> > It was fine on previous build where I could connect all 3 perfectly.
> 
> This is exactly what the `settsi_disable.flag` mentioned in the release notes above is for. Your knockoff controller will be using a different bluetooth chipset that doesn't like the SetTsi vendor command sent by the console. In older versions I was disabling this command for _all_ non-nintendo controllers, so your controller would have been happy. It turned out that faking the success of this command could create issues with input delays and degradation of WiFi speeds, so now I only disable it when a controller doesn't support it. In the case of DS4 controllers, the authentic ones _do_ support the command so we want to send it, but the third party controllers that don't support the command are indistinguishable from the real ones in software so we need to flag them manually.

Hi, i have the same problem with my DS4 when i unplug my joycons, and the rumble or vibration function doesn't work in-game
How can i create the .flag archive? do i need to write something in this?

**Reply by ndeadly** [reply]
You create an empty file called `settsi_disable.flag` in `sdmc:/config/MissionControl/controllers/<controller address>` where `<controller address>` is the 12-digit hex representation of you controller's mac address.

However if this is occuring when you unplug joycons this might not be the same issue. Controllers that are incompatible with the command I speak about will be disconnected by themselves shortly after connecting.

**Reply by Zohan1994** [reply]
Ohh i know now, my joystick doesn't disconnect but it won't stop vibrating when i unplug my joycons and pair with the console at player 1-2, and in-game the vibration doesn't work too.
With mission 0.5.0 beta i don't have this problem.
what should i do?

**Reply by ndeadly** [reply]
Only thing you can do for now is disable the rumble in the sysmodule [configuration](https://github.com/ndeadly/MissionControl#module-configuration). Nothing has changed in the rumble code since I released it, this is a super strange issue that seems to be related to the 13.1.0 update. I'm looking into it but I currently don't understand why it's happening.

**Reply by Zohan1994** [reply]
Okay i try to disable it :), maybe is a problem with firmware 13.1.0 or atmos last update. Thanks for reply faster 💪🏽

**Reply by ndeadly** [reply]
The rumble issues are now fixed in [v0.6.2](https://github.com/ndeadly/MissionControl/releases/tag/v0.6.2)
